app/grandchallenge/hanging_protocols/models.py

```diff
@@ -395,5 +398,9 @@ def check_all_interfaces_in_view_content_exist(self):
                     f"{', '.join(i.slug for i in undisplayable_interfaces)}"
                 )
 
+    def clean_view_content(self, view_content, hanging_protocol):
```

Comment:
```suggestion
    @staticmethod
    def clean_view_content(*, view_content, hanging_protocol):
```

Reply:
I kept the wrapping function `clean_view_content` as a method, so it can be called with `self.instance` in the form and still the validation functions inside it can still be called with `self.`
Otherwise it would be:

```python
@staticmethod
def clean_view_content(*, view_content, hanging_protocol):
       HangingProtocolMixin.check_consistent_viewports(view_content, hanging_protocol)
       HangingProtocolMixin.check_all_interfaces_in_view_content_exist(view_content)

```
And the call in the form will be:
`HangingProtocolMixin.clean_view_content(...
`

Is that what we want here?

Reply:
yeah true.
I will stick to it.
Done.